File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mpp = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in mpp:
                return [i, mpp[diff]]
            mpp[nums[i]] = i
        return [-1,-1]

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mpp;
        for (int i = 0;i < nums.size();i++) {
            int remaining = target - nums[i];
            if (mpp.find(remaining) != mpp.end()) return {i, mpp[remaining]};
            mpp[nums[i]] = i; 
        }
        return {-1,-1};
    }
};